src/DataViews/FunctionsDataView.cpp

```diff
@@ -246,76 +244,50 @@ void FunctionsDataView::OnContextMenu(const std::string& action, int menu_index,
 void FunctionsDataView::DoFilter() {
   m_FilterTokens = absl::StrSplit(absl::AsciiStrToLower(filter_), ' ');
 
-#ifdef WIN32
-  ParallelFilter();
-#else
-  // TODO: port parallel filtering
-  std::vector<uint64_t> indices;
-  const std::vector<const FunctionInfo*>& functions(functions_);
-  for (size_t i = 0; i < functions.size(); ++i) {
-    const FunctionInfo* function = functions[i];
-    std::string name =
-        absl::AsciiStrToLower(orbit_client_data::function_utils::GetDisplayName(*function)) +
-        orbit_client_data::function_utils::GetLoadedModuleName(*function);
-
-    bool match = true;
-
-    for (std::string& filter_token : m_FilterTokens) {
-      if (name.find(filter_token) == std::string::npos) {
-        match = false;
-        break;
-      }
-    }
+  const size_t number_of_threads = thread_pool_->GetPoolSize();
+  const size_t number_of_functions_per_thread = functions_.size() / number_of_threads;
```

Comment:
'number_of_functions_per_thread' can be 0 if we have less functions than threads. I guess this kinda works given the `if (thread + 1 == number_of_threads) end = functions_.size();` condition in the for loop, but it's convoluted. Also, in that case, we are going to schedule multiple jobs when only one is needed.

Reply:
Yes, I agree. It's rather convoluted. I wanted to avoid having a separate code-path for the single-thread-case - I'm not sure though if that's the best decision. WDYT?

Yes, `if (thread + 1 == number_of_threads) end = functions_.size();` makes all functions be checked in the "last" thread in case there a fewer functions than threads in the pool. And there is also a `if (end == begin) continue;` which avoids scheduling "empty" threads.

Now that I'm thinking about it, maybe I should introduce a minimum number of functions per thread. Even though tasks are rather lightweight, filtering just a couple of functions is probably way cheaper than the overhead introduced by the task scheduling. WDYT?

Reply:
To me this seems like a fine algorithm to split a vector to process across threads, I don't think it's that convoluted, but maybe you can add a comment on those two lines to make it clearer.

> maybe I should introduce a minimum number of functions per thread.

On the other hand now that you bring this up I think this should be done.

Reply:
I think evenly scheduling or evenly with the remainder in the last thread (as done here) is both fine. A minimum per thread is also fine, but I would not go to great length to optimize for the case of few functions since it will be fast anyway.

+1 for a just adding a few comments.

Reply:
There is also a risk that the last thread will do almost twice as much work as the other threads are doing in the case which cancel some of the effect of parallelization. Lets say we have 499 functions, thread 1-3 will process 100 functions each, thread 4 will process 199.

Reply:
Ah no nevermind, that's not right.

Reply:
I don't know, the algorithm obviously works, but for cases where processing of elements is more costly than a simple string comparison, proper balancing becomes more important. It would be cool eventually to have some utility that optimally balances the load for the number of threads available to minimize latency.

In this case, my surprise was from the realization that `number_of_functions_per_thread` can be zero, which intuitively doesn't seem right, and from the fact that we can spawn tasks that do nothing. A comment acknowledging those cases would be ok.

Reply:
I added a minimum which limits how small tasks get and also added a second thread pool.